bin/crew

```diff
@@ -909,6 +909,17 @@ def prepare_package(destdir)
     system 'find . -type l >> ../filelist'
     system 'cut -c2- ../filelist > filelist'
 
+    # check for FHS3 compliance
+    @fhs_compliant_prefix = ['bin', 'etc', 'include', 'lib', 'libexec', 'opt', 'sbin', 'share', 'var']
+    @fhs_compliant_prefix.append(ARCH_LIB) if ARCH_LIB == 'lib64'
+    Dir.foreach(CREW_DEST_PREFIX) do |filename|
+      next if filename == '.' or filename == '..'
+      unless @fhs_compliant_prefix.include?(filename)
+        puts "Unable to complete this build because #{CREW_PREFIX}/#{filename} in #{@pkg.name} is not FHS3 compliant.".lightred
+        abort unless ENV['CREW_FHS_NONCOMPLIANCE_ONLY_ADVISORY'] == '1'
```

Comment:
Would it make sense to abort outside the foreach loop in case more than one directory is non-compliant?

Reply:
That is a very good point, yes it would.

Reply:
I can't seem to figure it out, could you or @satmandu give me some example/kickstart code?

Reply:
Just move the abort line outside the loop:
```
    Dir.foreach(CREW_DEST_PREFIX) do |filename|
      next if filename == '.' or filename == '..'
      unless @fhs_compliant_prefix.include?(filename)
        puts "Unable to complete this build because #{CREW_PREFIX}/#{filename} in #{@pkg.name} is not FHS3 compliant.".lightred
      end
    end
    abort unless ENV['CREW_FHS_NONCOMPLIANCE_ONLY_ADVISORY'] == '1'
```

Reply:
Could we report all problems before aborting?

It sucks on long compiles to fix one thing, and then have it error out during the next compile.

Reply:
so abort at the end of the function? on it.

Reply:
0c48482